packages/drawnix/src/plugins/freehand/with-freehand-delete.ts

```diff
@@ -0,0 +1,85 @@
+import {
+    PlaitBoard,
+    Point,
+    Transforms,
+    toHostPoint,
+    toViewBoxPoint,
+} from '@plait/core';
+import { isDrawingMode } from '@plait/common';
+import { getFreehandPointers, isHitFreehand } from './utils';
+import { Freehand, FreehandShape } from './type';
+
+export const withFreehandDelete = (board: PlaitBoard) => {
+    const { pointerDown, pointerMove, pointerUp, globalPointerUp } = board;
+
+    let isErasing = false;
+    const deletedElements = new Set<string>();
+
+    const checkAndDeleteFreehandElements = (point: Point) => {
+        const viewBoxPoint = toViewBoxPoint(board, toHostPoint(board, point[0], point[1]));
+
+        const freehandElements = board.children.filter((element) =>
+            Freehand.isFreehand(element)
+        ) as Freehand[];
+
+        freehandElements.forEach((element) => {
+            if (!deletedElements.has(element.id) && isHitFreehand(board, element, viewBoxPoint)) {
+                deletedElements.add(element.id);
+                const elementIndex = board.children.findIndex(child => child.id === element.id);
+                if (elementIndex !== -1) {
+                    Transforms.removeNode(board, [elementIndex]);
```

Comment:
Maybe you can call the `CoreTransforms.removeElements(board, elements);` to remove multiple elements in final.

It will handle the path issue that multiple elements need be deleted in sync period, after the previous element is deleted the next element will occur path conflict.